lib/pbench/agent/results.py

```diff
@@ -102,100 +122,78 @@ def make_result_tb(self):
-            sys.exit(1)
-        self.tarball_md5 = Path(f"{tarball}.md5")
-        if not self.tarball_md5.exists():
-            logger.error("tarball's .md5 does not exist, '{}'", self.tarball_md5)
-            sys.exit(1)
+            raise FileNotFoundError(f"Tarball does not exist, '{self.tarball}'")
         self.logger = logger
-        server_rest_url = config.results.get("server_rest_url")
+        server_rest_url = config.get("results", "server_rest_url")
         self.upload_url = f"{server_rest_url}/upload/ctrl/{controller}"
 
-    def read_in_chunks(self, file_object):
+    def read_in_chunks(self, file_object: IO) -> Iterator[AnyStr]:
         data = file_object.read(self.chunk_size)
         while data:
             yield data
             data = file_object.read(self.chunk_size)
 
-    def copy_result_tb(self):
-        files = [f for f in self.tarball.parent.iterdir() if f.is_file()]
-        file_count = len(files)
-        if file_count != 2:
-            self.logger.error(
-                "(internal): unexpected file count, {}, associated with tarball, '{}'",
-                file_count,
-                self.tarball,
-            )
-            sys.exit(1)
-
-        with open(self.tarball_md5, "r") as md5fp:
-            md5sum = md5fp.read().split()[0]
-        filename = secure_filename(str(self.tarball))
-        headers = {"filename": filename, "Content-MD5": md5sum}
+    def copy_result_tb(self, token: str) -> None:
+        """copy_result_tb - copies tb from agent to configured server upload URL
+
+            Args
+                token -- an Authorization key which will allow the user to be
+                    authorized by the server.
+        """
+        headers = {
+            "filename": secure_filename(str(self.tarball)),
+            "Content-MD5": md5sum(self.tarball),
+            "Authorization": f"Bearer {token}",
+        }
         with self.tarball.open("rb") as f:
             try:
                 response = requests.put(
                     self.upload_url, data=self.read_in_chunks(f), headers=headers
                 )
                 self.logger.info("File uploaded successfully")
-            except Exception:
-                self.logger.exception("There was something wrong with your request")
-                sys.exit(1)
-        if not response.status_code == 200:
-            self.logger.error(
-                "There was something wrong with your request, error code: '{}'",
-                response.status_code,
+            except requests.exceptions.ConnectionError:
+                raise RuntimeError(f"Cannot connect to '{self.upload_url}'")
+            except Exception as exc:
+                raise FileUploadError(
+                    "There was something wrong with file upload request:  "
+                    f"file: '{self.tarball}', URL: '{self.upload_url}', ({exc})"
+                )
+        if not response.ok:
```

Comment:
Conversely, with `raise_for_status()` I don't think we should be falling through unless it _was_ OK.

Reply:
Right; will try that.